source/language/types.rst

```diff
@@ -267,6 +267,38 @@ are all of type ``float[64]``.
 
 Note that `e` is a valid identifier. `e/E` are also used in scientific notation where appropriate.
 
+Mathematical functions available for constant initialization
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In addition to simple arithmetic functions used in expressions initializing constants,
+OpenQASM 3 offers the following built-in mathematical operators followed by
+their argument expression in parentheses:
+
+.. container::
+   :name: tab:built-in-math
+
+   .. table:: Built-in mathematical functions in OpenQASM3.
+
+      +--------+
+      | arccos |
+      +--------+
+      | arcsin |
+      +--------+
+      | arctan |
+      +--------+
+      | cos    |
+      +--------+
+      | exp    |
+      +--------+
+      | ln     |
```

Comment:
The above paragraph also mentions the function `log`.  We should define what the logarithm bases are.  I think in keeping with our verbose approach, perhaps we should have `log` as base-e log, not have `ln`, and if there's a very strong use-case for them, call base-10 or base-2 log `log10` or `log2` respectively.  They shouldn't generally be necessary, though, since `log(x) / log(y)` is the logarithm of `x` in base `y` in no matter what base `log` is.

The minimal expected overload would be `log(float) -> float`.  Conceivably, we could allow `log(complex) -> complex` as well, in which case we need to mathematically define how the branch cut is handled.